src/librustc/traits/trans/mod.rs

```diff
@@ -202,9 +202,12 @@ impl<'gcx> DepTrackingMapConfig for ProjectionCache<'gcx> {
         let def_ids: Vec<DefId> =
             key.walk()
                .filter_map(|t| match t.sty {
-                   ty::TyAdt(adt_def, _) => Some(adt_def.did),
-                   ty::TyProjection(ref proj) => Some(proj.trait_ref.def_id),
-                   _ => None,
+                    ty::TyAdt(adt_def, _) => Some(adt_def.did),
+                    ty::TyProjection(ref proj) =>
+                        // FIXME(tschottdorf): prof.item_def_id might be good enough?
+                        // Either way, use of tls::with not appropriate here.
```

Comment:
In short, as the comment explains, we just need some consistent function that maps the same predicate to the same dep-node.